pandas/core/arrays/arrow/array.py

```diff
@@ -521,6 +525,48 @@ def dropna(self: ArrowExtensionArrayT) -> ArrowExtensionArrayT:
         else:
             return type(self)(pc.drop_null(self._data))
 
+    @doc(ExtensionArray.fillna)
+    def fillna(
+        self: ArrowExtensionArrayT,
+        value: object | ArrayLike | None = None,
+        method: FillnaOptions | None = None,
+        limit: int | None = None,
+    ) -> ArrowExtensionArrayT:
+
+        value, method = validate_fillna_kwargs(value, method)
+
+        try:
+            if method is None and limit is None:
+                if is_array_like(value):
+                    val = cast(ArrayLike, value)
+                    if len(val) != len(self):
+                        raise ValueError(
+                            f"Length of 'value' does not match. Got ({len(val)}) "
+                            f" expected {len(self)}"
+                        )
+                    if not isinstance(val, (pa.Array, pa.ChunkedArray)):
+                        val = pa.array(val, type=self._data.type, from_pandas=True)
+                else:
+                    val = pa.scalar(value, type=self._data.type, from_pandas=True)
+                return type(self)(pc.fill_null(self._data, fill_value=val))
+
+            elif method == "pad" and limit is None and not pa_version_under7p0:
+                return type(self)(pc.fill_null_forward(self._data))
+
+            elif method == "backfill" and limit is None and not pa_version_under7p0:
+                return type(self)(pc.fill_null_backward(self._data))
+
+        except (pa.ArrowInvalid, pa.ArrowTypeError):
```

Comment:
Can these `try/excepts` be more tightly scoped to where these exceptions are expected (guessing the `pa.array` and `pc.fill_null*` calls

Reply:
yep, good point. updated to more tightly scope